packages/web-forms/src/directives/FullWidth.ts

```diff
@@ -0,0 +1,22 @@
+interface FullWidthBinding {
+	value: boolean | undefined;
+}
+
+const checkWidth = (el: HTMLElement, binding: FullWidthBinding) => {
+	if (binding.value === false) return;
+
+	const items: Element[] = Array.from(el.children);
+
+	if (items.find((item) => item.scrollWidth * 2 > el.parentElement!.offsetWidth)) {
+		el.classList.add('full-width');
+	} else {
+		el.classList.remove('full-width');
+	}
+};
```

Comment:
So... when we discussed this on chat the other day, we talked about maybe holding off if it feels like it's not doing what we want. After playing with it some, I feel pretty strongly that that's the best option. The behavior as implemented is surprising in a lot of ways:

- It's based entirely on comparing measurements at the time of mount, which are determined in part by the available space at that same time
- It adjusts appearance based on media query, rather than actual space used by the options
- It does not adjust based on resizing (which alleviates my performance concern, but doesn't seem to do what I'd expect)
- If resizing from a smaller viewport to larger, it makes otherwise narrow options full width in ways I wouldn't expect
- I didn't test it, but I don't expect it will adjust based on form language change

I think it might be worth holding off on this for now, and coming back to it with fresh eyes if it feels strictly necessary. If so, maybe we can pair on it and think about some of these nuances together.

Reply:
The directive is listening to mount and update events, so it works with language change as well. I didn't handle resizing event, it felt more like a developer thing than end user use case.

In any case, I have removed the directive and created a separate issue to tackle this https://github.com/getodk/web-forms/issues/156. Yes, pairing on it will be helpful.